### PR TITLE
Documentation update for playbook feature

### PR DESCRIPTION
**Pull request by fergven**
Updated documentation for new playbook feature.

### CODE REVIEW

File: README-PLAYBOOK.md [hunk 1]
```diff
@@ -1,164 +1,164 @@
...
-| url         | string        ||
+| Field         | Type          | Description |
+|---|---|---|
+| credentials | `Credentials` | A [Credential](#credential) object that defines the credentials used to connect to the selected provider `platform`. |
+| trustBundle | string        | Defines path to PEM-formatted trust bundle that contains the root (and optionally intermediate certificates) to use to trust the TLS connection. If omitted, will attempt to use operating system trusted CAs. |
+| platform    | string        | For TLS Protect Datacenter, either `tpp` or `tlspdc`. For TLS Protect Cloud, either `vaas` or `tlspc`. |
+| url         | string        | ***REQUIRED*** when platform is `tpp` or `tlspdc`. *OPTIONAL* when platform is `vaas` or `tlspc` (defaults to api.venafi.cloud). If url string does not include `https://`, it will be added automatically. For connection to TLS Protect Datacenter, `url` must include the full API path (for example `https://tpp.company.com/vedsdk/`. |
 
-### Credentials
+### Credential
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:84]: The object here is Credentials (plural) as stated in line 79

File: README-PLAYBOOK.md [hunk 2]
```diff
@@ -1,164 +1,164 @@
...
-|--------------|------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| accessToken  | `Credentials`          ||
-| apikey       | string                 ||
-| clientId     | string                 ||
-| pkcs12       | `CertificateTask.Name` | Use a configured certificateTask (by name) to reference a certificate to be used for certificate authentication. Only supported by TPP endpoints. Referenced certificateTask must have an installation of type `pkcs12` |
-| refreshToken | string                 ||
-| scope        | string                 ||
+| Field        | Type                              | Description |
+|---|---|---|
+| apikey       | string     | ***REQUIRED*** - when [Connection.platform](#connection) is `vaas`. *IGNORED* otherwise. |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:88]: `apikey` should be `apiKey` with an upper case K

File: README-PLAYBOOK.md [hunk 3]
```diff
@@ -1,164 +1,164 @@
...
+| accessToken  | string     | *OPTIONAL* - Used when [Connection.platform](#connection) is `tpp` for authenticating to the REST API. If omitted, invalid, or expired, vcert will attempt to use the [Credential.p12Task](#credential) or [Credential.refreshToken](#credential) to get a valid accessToken. Upon successful refresh, this value will be overwritten with the new valid accessToken. *IGNORED* when [Connection.platform](#connection) is `vaas` or `tlspc`. |
+| clientId     | string     | *OPTIONAL* - Used when [Connection.platform](#connection) is `tpp` to map to the API integration to be used. If omitted, uses `vcert-cli` as default. *IGNORED* when [Connection.platform](#connection) is `vaas` or `tlspc`. |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:90]: default value for cliendId is `vcert-sdk` not `vcert-cli`

File: README-PLAYBOOK.md [hunk 4]
```diff
@@ -1,164 +1,164 @@
...
+| backupFiles          | boolean | *OPTIONAL* - When `true`, backup existing certificate files before replacing during a renewal operation. Defaults to `false`. |
+| file                 | string  | ***REQUIRED*** when [Installation.format](#installation) is `PKCS#12`, `PEM`, or `JKS`. Specifies the file path and name for the certificate file (PEM) or PKCS#12 / JKS bundle. Example `/etc/ssl/certs/myPEMfile.cer`, `/etc/ssl/certs/myPKCS12.p12`, or `/etc/ssl/certs/myJKS.jks`. *IGNORED* for other values of [Installation.format](#installation). |
+| jksAlias             | string  | ***REQUIRED*** when [Installation.format](#installation) is `JKS`. Specifies the certificate alias value within the Java Keystore. *IGNORED* when [Installation.format](#installation) is not `JKS`. |
+| jksPassword          | string  | ***REQUIRED*** when [Installation.format](#installation) is `JKS`. Specifies the password for the JKS. *IGNORED* when [Installation.format](#installation) is not `JKS`. |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:115]: Change JKS acronym here for Java Keystore to be consistent with jksAlias' description:
`Specifies the password for the Java Keystore`

File: README-PLAYBOOK.md [hunk 5]
```diff
@@ -1,164 +1,164 @@
...
+| chainFile            | string  | ***REQUIRED*** when [Installation.format](#installation) is `PEM`. Specifies the file path and name for the chain PEM bundle (Example `/etc/ssl/certs/myChain.cer`). *IGNORED* for other values of [Installation.format](#installation) |
+| keyFile              | string  | ***REQUIRED*** when [Installation.format](#installation) is `PEM`. Specifies the file path and name for the private key PEM file (Example `/etc/ssl/certs/myKey.key`). *IGNORED* for other values of [Installation.format](#installation) |
+| location             | string |  ***REQUIRED*** when [Installation.format](#installation) is `CAPI`. Specifies the Windows CAPI store to place the installed certificate. Typically `"LocalMachine\My"` or `"CurrentUser\My"`. *IGNORED* if [Installation.format](#installation) is NOT `CAPI`. **NOTE:** if the location is not contained within `"`, the backslash `\` must be properly escaped (i.e. `LocalMachine\\My`). |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:118]: regarding the quotes and double quotes, it is the opposite:
if location is in double quotes the backslash must be escaped:
`location: "LocalMachine\\My"`

if the location is in single quotes or no quotes, the backslash does not need escaping:
`location: 'LocalMachine\My'`

File: README-PLAYBOOK.md [hunk 6]
```diff
@@ -1,164 +1,164 @@
...
-in CAPI. Tool can run any after install script as well. 
-2. Remote machines like ATMs, can use this tool to pull provisioning certificates on startup or regular intervals. Once 
-the certificate is received, it can run any post-processing scripts configured in the tool.
+VCert Playbook functionality solves the "last mile" problem. VCert has historically done a great job fetching certificates, but getting those certificates installed in the right location usually required custom scripting. The Playbook functionality addresses this use case natively. 
+
+> Throughout this article, we use _TLS Protect Datacenter_ to refer to Trust Protection Platform and _TLS Protect Cloud_ to refer to Venafi as as Service.
+
+## Key features of VCert playbook
+
+- You don't have to pass long command line arguments. You can now call VCert with playbook YAML files, which helps with automation and maintenance.
```
Comment (jdw2465VEN) [inline README-PLAYBOOK.md:9]: Here's a suggestion to make this more scannable, which customers love; also some minor edits for clarity:

- **Simplified Commands**: With VCert Playbook, you can avoid long command-line arguments. Instead, use playbook YAML files with VCert to enhance automation and maintenance ease.
  
- **Flexible Certificate Placement**: You can designate where to place the certificate and the format in which you want it once received from Venafi. VCert Playbook supports common keystore formats like PEM, JKS, PKCS#12, accommodating folder locations and the Windows CAPI store.
  
- **Post-Installation Actions**: Specify any actions that must be carried out after the certificate is installed. This includes restarting services like Apache or Nginx, or running any other scripts needed once the certificate is in the right location.
  
- **Smart Certificate Renewals**: VCert Playbook checks if a certificate already exists and whether it's due for renewal before requesting a new one. This functionality lets you run a script regularly without unnecessarily renewing certificates.
  
- **Compatibility**: VCert Playbook works seamlessly with both TLS Protect Cloud and TLS Protect Datacenter, ensuring it fits your particular environment.

File: README-PLAYBOOK.md [hunk 7]
```diff
@@ -1,164 +1,164 @@
...
+
+- You don't have to pass long command line arguments. You can now call VCert with playbook YAML files, which helps with automation and maintenance. 
+- You can specify where the certificate goes, and in which format, once received from Venafi. It supports common keystore formats (PEM, JKS, PKCS#12) in folder locations as well as Windows CAPI store. 
+- You can specify any post-installation actions need to be taken, such as restarting services (apache, nginx) or any other script you would like to run after the certificate is available in the right location. 
+- For certificate renewals, it checks to see if the certificate already exists and if it is due for renewal before requesting a new certificate. This allows you to run a script regularly without renewing certificates that don't need to be.
+- VCert Playbook functionality works with both TLS Protect Cloud and TLS Protect Datacenter.
+
+## Example use cases
+- Renewal parameters can be setup to automatically renew the certificate before it expires. This model assumes that VCert is added to a daily cronjob or is executed on a regular basis by some other automation. Default renewal is 10% of remaining certificate lifetime.
```
Comment (jdw2465VEN) [inline README-PLAYBOOK.md:16]: Same suggestion here and again with some suggested edits:

- **Automated Certificate Renewal**: You can set renewal parameters to have VCert automatically renew certificates before expiration. This approach assumes that VCert is part of a daily cronjob or is executed routinely through other automation methods. By default, renewal occurs at 10% of the remaining certificate lifetime.

- **Effortless API Access Updates**: When using TLS Protect Datacenter, VCert will automatically update API access and refresh tokens within the playbook. This feature ensures continuous operation without manual intervention. It leverages a refresh token to acquire a new access token when needed, an approach that's particularly effective when paired with a long-lasting refresh/grant token and a short-lived access token, such as a 3-year refresh token and a 1-hour access token.

File: README-PLAYBOOK.md [hunk 8]
```diff
@@ -1,164 +1,164 @@
...
-| Field   | Type      | Description |
-|---------|-----------|-------------|
-| type    | string    ||
-| name    | string    ||
-| value   | string    ||
-
+|---|---|---|
+| chain           | string                 | *OPTIONAL* - Determines the ordering of certificates within the returned chain. Valid options are `root-first`, `root-last`, or `ignore`. Defaults to `root-last`. |
+| csr             | string                 | *OPTIONAL* - Specifies where the CSR and PrivateKey are generated: use `local` to generate the CSR and PrivateKey locally, or `service` to have the PrivateKey and CSR generated by the specified [Connection.platform](#connection). Defaults to `local`. |
+| fields          | array of strings)      | *OPTIONAL* - Sets the specified custom field on certificate object. Only valid when [Connection.platform](#connection) is `tpp`. |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:128]: This is not an array of Strings, it is an array of `CustomField` object. Which I think is missing in the doc:

### CustomField

| Field   | Type      | Description |
|---------|-----------|-------------|
| name    | string    ||
| value   | string    ||

File: README-PLAYBOOK.md (hunk 6 above)
Comment (jdw2465VEN) [reply README-PLAYBOOK.md:9]: I should have made the bolded text sentence case, per our style guide

File: README-PLAYBOOK.md [hunk 9]
```diff
@@ -1,164 +1,164 @@
...
+| nickname        | string                 | *OPTIONAL* - Specify the certificate object name to be created in TPP for the requested certificate. If not specified, TPP will use the [Subject.commonName](#subject). Only valid when [Connection.platform](#connection) is `tpp`.|
+| sanIP           | array of string        | *OPTIONAL* - Specify one or more IP SAN entries for the requested certificate. |
+| issuerHint      | string                 | *OPTIONAL* - Used only when [Request.validDays](#request) is specified to determine the correct Specific End Date attribute to set on the TPP certificate object. Valid options are `DIGICERT`, `MICROSOFT`, `ENTRUST`, `ALL_ISSUERS`. If not defined, but `validDays` are set, the attribute 'Specific End Date' will be used. Only valid when [Connection.platform](#connection) is `tpp`. |
+| keyCurve        | string                 | ***REQUIRED*** when [Request.keyType](#request) is `ECDSA`, `EC`, or `ECC`. Valid values are `P256`, `P384`, `P521`, `ED25519`. |
+| keySize         | integer                | *OPTIONAL* - Specifies the key size when specified [Request.keyType](#request) is `RSA`. Defaults to 2048. |
```
Comment (rvelaVenafi) [inline README-PLAYBOOK.md:135]: supported values for keySize are: `1024`, `2048`, `4096`, `8192`

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]
Added comments regarding the documentation

**Review by marcos-albornoz — APPROVED** [review]

**Review by luispresuelVenafi — APPROVED** [review]

**Review by fergven — COMMENTED** [review]
Changes made

**Review by fergven — COMMENTED** [review]
Changes made

**Review by fergven — COMMENTED** [review]
Changes made

**Comment by fergven** [comment]
> Added comments regarding the documentation

I've made all the changes.

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**